backend/Controllers/BookingController.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using CinemaApp.Models;
+
+namespace CinemaApp.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class BookingController : ControllerBase
+    {
+        private readonly BookingContext _context;
+
+        public BookingController(BookingContext context)
+        {
+            _context = context;
+        }
+
+        // GET: api/Bookings
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<Booking>>> GetBookings()
+        {
+            return await _context.Bookings.ToListAsync();
+        }
+
+        // GET: api/Bookings/5
+        [HttpGet("{id}")]
+        public async Task<ActionResult<Booking>> GetBooking(int id)
+        {
+            var booking = await _context.Bookings.FindAsync(id);
+
+            if (booking == null)
+            {
+                return NotFound();
+            }
+
+            return booking;
+        }
+
+        // PUT: api/Bookings/5
+        // To protect from overposting attacks, enable the specific properties you want to bind to, for
+        // more details, see https://go.microsoft.com/fwlink/?linkid=2123754.
+        [HttpPut("{id}")]
+        public async Task<IActionResult> PutBooking(int id, Booking booking)
+        {
+            if (id != booking.ID)
+            {
+                return BadRequest();
+            }
+
+            _context.Entry(booking).State = EntityState.Modified;
```

Comment:
This should not be the case. If you update an entity in the db, you only need to set the updated fields and call the save mechanism. It will handle the state by itself. Again, you should not work directly with db entities.